subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -259,7 +266,7 @@ static void sm_handle_timeout_state(struct lwm2m_message *msg,
 #endif
 	{
 		if (client.engine_state == ENGINE_REGISTRATION_SENT) {
-			event = LWM2M_RD_CLIENT_EVENT_REGISTRATION_FAILURE;
+			event = LWM2M_RD_CLIENT_EVENT_NETWORK_ERROR;
```

Comment:
> LWM2M_RD_CLIENT_EVENT_REGISTRATION_FAILURE should be only reported case when server reject by response registration.

Why? How it's different from `LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE` when both timeout and server rejectoin are handled the same? If it's needed to differentiate between timeout/rejection, perhaps a new event should be introduced?

Reply:
I was thinking to use that. I was seeing that server reject and timeout use same event but. Server rejected enter idle state but timeout continue healing at engine side. I think it make sense to use LWM2M_RD_CLIENT_EVENT_REG_UPDATE_FAILURE